src/Aspire.Hosting.AWS/CDK/CDKExtensions.cs

```diff
@@ -0,0 +1,229 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Amazon.CDK;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.AWS;
+using Aspire.Hosting.AWS.CDK;
+using Aspire.Hosting.AWS.CloudFormation;
+using Constructs;
+using Environment = System.Environment;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Extension methods for adding AWS CDK as a provisioning resources.
+/// </summary>
+public static class CDKExtensions
+{
+    /// <summary>
+    /// Adds an AWS CDK stack as resource. The CloudFormation stack name will be the resource name prefixed with 'Aspire-'
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the stack resource.</param>
+    public static IResourceBuilder<IStackResource> AddAWSCDKStack(this IDistributedApplicationBuilder builder, string name)
+        => AddAWSCDKStack(builder, name, "Aspire-" + name);
+
+    /// <summary>
+    /// Adds an AWS CDK stack as resource.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the stack resource.</param>
+    /// <param name="stackName">Cloud Formation stack same if different from the resource name.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<IStackResource> AddAWSCDKStack(this IDistributedApplicationBuilder builder, string name,
+        string stackName)
+    {
+        builder.AddAWSProvisioning();
+        var resource = new StackResource(name, new Stack(ResolveCDKApp(builder), stackName));
+        return builder
+            .AddResource(resource)
+            .WithInitialState(new()
+            {
+                Properties = [],
+                ResourceType = GetResourceType<Stack>(resource),
+            })
+            .WithManifestPublishingCallback(resource.WriteToManifest);
+    }
+
+    /// <summary>
+    /// Adds and build an AWS CDK stack as resource.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="stackBuilder">The stack builder delegate.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<IStackResource<T>> AddAWSCDKStack<T>(this IDistributedApplicationBuilder builder,
```

Comment:
@normj  I agree that adding `string? stackName = null` is a better solution. However, when I do this I get `'AddAWSCDKStack' violates the backcompat requirement: 'API with optional parameter(s) should have the most parameters amongst its public overloads'`. For this reason there is a separate overload. The method with the `ConstructBuilderDelegate<T>` is a generic method and for this reason both can't be merged. An alternative would be to create a `AddAWSCDKStackBuilder<T>` method instead.

Reply:
The `AddAWSCDKStackBuilder<T>` doesn't need a `stackName` as it will be taken from the construct as shown in the playground example.
`var customStack = builder.AddAWSCDKStack("custom", scope => new CustomStack(scope, "Aspire-custom"));`